app/Models/Concerns/HasMemberships.php

```diff
@@ -0,0 +1,148 @@
+        return $this->memberships()
+            ->secondary()
+            ->orderByPriority();
+    }
+
+    public function updatePrimaryMembership(string $division, string $region): bool
+    {
+        $matchingMembership = resolve(Membership::class)->findPrimaryByVATSIMLocality($division, $region);
+
+        if (! $matchingMembership) {
+            return false;
+        }
+
+        // Check if the user already has this membership, and the division/region combination is the same
+        if ($this->hasMembership($matchingMembership) && ($this->primaryMembership()->pivot->division == $division && $this->primaryMembership()->pivot->region == $region)) {
+            return false;
+        }
+
+        // Check we can have secondary memberships
+        if (! $matchingMembership->can_have_secondaries) {
+            $this->removeSecondaryMemberships();
+        }
+
+        // End Previous Primary Membership
+        if ($currentMembership = $this->primaryMembership()) {
+            $this->removeMembership($currentMembership);
+        }
+
+        $this->memberships()->attach($matchingMembership, [
+            'division' => $division,
+            'region' => $region,
+            'started_at' => Carbon::now(),
+        ]);
+
+        return true;
+    }
+
+    public function addSecondaryMembership(Membership $membership): bool
+    {
+        throw_if($membership->primary, new MembershipNotSecondaryException());
+
+        // Check we can have secondary memberships
+        if ($this->primaryMembership() && ! $this->primaryMembership()->can_have_secondaries) {
+            throw new PrimaryMembershipDoesntAllowSecondaryException();
+        }
+
+        // Check if the user already has this membership
+        if ($this->hasMembership($membership)) {
+            return false;
+        }
+
+        $this->memberships()->attach($membership);
+
+        return true;
+    }
+
+    public function removeSecondaryMemberships(): void
+    {
+        $this->secondaryMemberships->each(function ($membership) {
+            $this->removeMembership($membership);
+        });
+    }
+
+    /**
+     * @param Membership|int|string $membership
+     * @return bool
+     */
+    public function hasMembership($membership): bool
+    {
+        if ($membership instanceof Membership) {
+            return $this->memberships()->where('memberships.id', $membership->id)->exists();
+        } elseif (is_string($membership)) {
+            // Assume is an identifier
+            return $this->memberships()->where('memberships.identifier', $membership)->exists();
+        }
+        // Assume is an ID
+        return $this->memberships()->where('memberships.id', $membership)->exists();
+    }
+
+    public function removeMembership(Membership $state): int
+    {
+        return $this->memberships()->updateExistingPivot($state, [
+            'ended_at' => Carbon::now(),
+        ]);
+    }
+
+    public function getPrimaryMembershipAttribute(): ?Membership
```

Comment:
Will this not conflict with the method of a similar name? (and why can't you just use the relationship?

Reply:
It's not a relationship it is a query. As such, `primaryMembership()` has to be called as a function, because it is indeed not a relationship. This attribute acts as an accessor to allow it to be returned as part of a API query.